configuration/src/main/java/io/airlift/configuration/ConfigurationFactory.java

```diff
@@ -148,12 +148,24 @@ public void registerConfigurationClasses(Module module)
 
     private <T> ConfigDefaults<T> getConfigDefaults(Key<T> key)
     {
-        List<ConfigDefaults<T>> defaults = registeredDefaultConfigs.get(key).stream()
+        Key generalDefaultsKey = Key.get(key.getTypeLiteral(), GeneralDefaults.class);
+        List<ConfigDefaults<T>> generalDefaults = registeredDefaultConfigs.get(generalDefaultsKey).stream()
                 .map(holder -> (ConfigDefaultsHolder<T>) holder)
                 .sorted()
                 .map(ConfigDefaultsHolder::getConfigDefaults)
                 .collect(toList());
 
+        List<ConfigDefaults<T>> keyDefaults = registeredDefaultConfigs.get(key).stream()
+                .map(holder -> (ConfigDefaultsHolder<T>) holder)
+                .sorted()
+                .map(ConfigDefaultsHolder::getConfigDefaults)
+                .collect(toList());
+
+        List<ConfigDefaults<T>> defaults = ImmutableList.<ConfigDefaults<T>>builder()
```

Comment:
If you declare this at the top of the method, you can change the stream processing above from `.collect(toList());` to `.forEach(defaults::add));`